kafka-webview-ui/src/main/java/org/sourcelab/kafka/webview/ui/configuration/WebServerConfig.java

```diff
@@ -0,0 +1,33 @@
+package org.sourcelab.kafka.webview.ui.configuration;
+
+import org.apache.catalina.connector.Connector;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.web.embedded.tomcat.TomcatServletWebServerFactory;
+import org.springframework.boot.web.server.WebServerFactoryCustomizer;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+/**
+ * Manages WebServer Configuration and Customization.
+ */
+@Configuration
```

Comment:
Is the idea that this secondary port will only serve metrics/actuator requests?

Reply:
So here we go with the issue/problem/pain-point I found working on this task.
Spring Boot Actuator has a property to set the listening port to another one, different than the normal server:
- `server.port: 8080` this is the normal server port
- `management.server.port: 9090` this is the new server port opened to serve just Spring Boot Actuator endpoints

The issue is between this management port and Spring Security.
If you connect to `localhost:8080/` and you are not yet logged-in, the server will present you the login page at `localhost:8080/login`. This works fine.
The same behaviour is expected for Actuator endpoints as well at `localhost:9090/actuator`, but once redirected to login page at `localhost:9090/login` comes the problem: 404 not found! This is because the normal server answers just on port 8080 and the Actuator server answers on port 9090.
I tried to find examples on internet or some reference in the Spring documentation without success.

For this reason I decided to leave normal and Actuator server on the same port, opening a secondary port. This way it depends only on the user decides to do, both 8080 and 9090 work the same. The login page is always presented properly and the redirection works perfectly.

I know this is not really an elegant solution, but for now I see just these alternatives:
1.  leave normal and Actuator server on the same port (e.g. 8080)
2. the solution currently implemented in this PR
3. search a way to hack into the Actuator Spring Filter Chain and pilot the redirection from port 9090 to port 8080 and back after login

Honestly I don't know how difficult and how helpful the documentation would be trying to realise alternative 3.

If you have a better way in mind, please let me know :)

Reply:
So what about adding something like the following:

```java
@Configuration
@Order(1)
public class ActuatorSecurity extends WebSecurityConfigurerAdapter {

    @Override
    protected void configure(HttpSecurity http) throws Exception {
        // Actuator uses http basic auth
        http
            // Not sure if we want to expose the health metrics w/o auth or not :shrug:
            .authorizeRequests()
            .antMatchers(HttpMethod.GET, "/actuator/health").permitAll()
            .and()
            .antMatcher("/actuator/**")
            .authorizeRequests()
            .anyRequest()
            .hasRole("ADMIN")
            .and()
            .httpBasic();

// Or more simplified...
//        http
//            .antMatcher("/actuator/**")
//            .authorizeRequests().anyRequest().authenticated().and()
//            .httpBasic();
    }
}
```

That should perform Http Basic Auth for requests that hit actuator.  If the management port is set to a different port, it will prompt for basic auth without the form login redirect.

Things seem to get a bit weird if you login to the application, and then hit the /actuator end point tho...you seem to get logged out of the application for reasons I don't quite understand.

Thoughts?

Reply:
> Things seem to get a bit weird if you login to the application, and then hit the /actuator end point tho...you seem to get logged out of the application for reasons I don't quite understand.

I totally agree on that. It's pretty weird but I found no other solutions before your suggestion! I had quite little experience with Spring Security, compared to the rest of Spring ecosystem, so thanks because learning a lot!
I think this is an elegant way to realise what we want to achieve!

> Thoughts?

I have just 2 comments:
- better avoid `@Order(1)` because it's always a better idea to load custom configuration AFTER Spring default configuration, especially talking about Spring Security
- to have a right behaviour in relation to Kubernetes, it's good practice to expose `/actuator/info`, `/actuator/health` and `/actuator/prometheus` endpoints (only GET) without login, you will see in the next PR about adding Kubernetes manifests and CoreOS Prometheus Operator manifests ;)

`INFO:` there is not problem exposing `/actuator/health` without login, because with the property `management.endpoint.health.show-details` set to `when_authorized`, you can have more info (disk-space, db, mail, etc.) just if you are logged-in :)

As you can see in my last commit, I introduced a new security config class dedicated to the Actuator. This class is loaded right before the security config for UI.

Reply:
Awesome!  Yeah likewise my SpringSecurity knowledge is pretty limited, and actuator is even less, so glad to have you to bounce ideas off of :)

Reply:
Glad to have found your project! It's awesome!

Solving this ;)